src/engine.rs

```diff
@@ -243,6 +250,27 @@ impl Reedline {
         res
     }
 
+    /// Return terminal support on keyboard enhancement
+    pub fn can_use_kitty_protocol(&mut self) -> bool {
+        if let Ok(b) = crossterm::terminal::supports_keyboard_enhancement() {
+            b
+        } else {
+            false
+        }
+    }
+
+    /// Enable keyboard enhancement to disambiguate escape code
+    pub fn enable_kitty_protocol(&mut self) -> Result<()> {
```

Comment:
Ah yes, I followed prior patterns. It might be best to just.. no return?

Reply:
> Ah yes, I followed prior patterns. It might be best to just.. no return?
>
this is why i said it might be a dumb question because i've never had a look at the source base of this repo so far...

looking at what the function does, it looks to me it could just be
```rust
pub fn enable_kitty_protocol(&mut self) {
    self. = true;
}
```
and if it's not an implementation of some trait :thinking: 

again, not sure if it should be removed or not, just looking locally at this particular function :relieved:

Reply:
I think I will stick with the Result despite some itch :laughing:

Reply:
I think we should only do this if necessary. The other commands directly write to STDOUT to perform their operation and are just fallible. As long as this is just setting an internal flag I think it should be infallible.

There may be an argument if we were to move the check for the flags here but as long we don't do it, let's not make preemptive API decisions. reedline is still quite liberal with making breaking changes if necessary. Until then lets keep the API as simple as possible.